Add depcheck for Unused Dependency Detection

sumanjeet0012 opened the pull request:
Partially Fixes: https://github.com/CheckerNetwork/roadmap/issues/221

I have integrated `depcheck` into the project to detect unused dependencies. Additionally, I have added a configuration file to exclude `@Types`, `TypeScript`, and `depcheck` from the analysis, as these may produce false positives.

juliangruber commented on .depcheckrc:
```diff
@@ -0,0 +1,5 @@
+ignores: [
+  "@types/*",    # Ignore TypeScript type definitions
+  "typescript",  # TypeScript itself is used by the compiler but not imported
```
```suggestion
```

I tested it without this line, `depcheck` doesn't complain about `typescript`

juliangruber commented on .depcheckrc:
```diff
@@ -0,0 +1,5 @@
+ignores: [
+  "@types/*",    # Ignore TypeScript type definitions
+  "typescript",  # TypeScript itself is used by the compiler but not imported
+  "depcheck",    # Used in npm scripts but not imported
```
```suggestion
```

I tested it without this line, `depcheck` doesn't complain about itself either

juliangruber commented on .depcheckrc:
```diff
@@ -0,0 +1,5 @@
+ignores: [
+  "@types/*",    # Ignore TypeScript type definitions
```
`depcheck` complains about `@types/pg`. If I remove that, everything still seems to be fine. It does _not_ complain about `@types/mocha`, so it's not that it in general doesn't understand types. I think we can remove this line as well, and thereby the whole file.

juliangruber commented on package.json:
```diff
@@ -9,11 +9,13 @@
         "test:types": "tsc -p .",
-        "release": "np"
+        "release": "np",
+        "check-deps": "depcheck"
```
```suggestion
        "test:deps": "depcheck"
```

juliangruber replied:
Could you please add `npm run test:deps` to the `"test"` script?

sumanjeet0012 replied:
Done

juliangruber replied:
> Could you please add npm run test:deps to the "test" script?

This is missing

sumanjeet0012 replied:
Now I added it.

pyropy reviewed (commented):
Thank you for putting all this effort @sumanjeet0012 ! 🎉 

Would you mind adding this dependency check step to [ci.yml](https://github.com/CheckerNetwork/spark-evaluate/blob/main/.github/workflows/ci.yml) file too?

sumanjeet0012 commented:
> Thank you for putting all this effort @sumanjeet0012 ! 🎉
> 
> Would you mind adding this dependency check step to [ci.yml](https://github.com/CheckerNetwork/spark-evaluate/blob/main/.github/workflows/ci.yml) file too?

Sure, Added dependency check in the Build Job.

sumanjeet0012 commented:
> > > Thank you for putting all this effort @sumanjeet0012 ! 🎉
> > > Would you mind adding this dependency check step to [ci.yml](https://github.com/CheckerNetwork/spark-evaluate/blob/main/.github/workflows/ci.yml) file too?
> > 
> > 
> > Sure, Added dependency check in the Build Job.
> 
> Great job! Looks like the CI is now failing due to unused dependencies
> 
> ```
> Run npm run test:deps
> 
> > @filecoin-station/spark-evaluate@1.0.4 test:deps
> > depcheck
> 
> Unused dependencies
> * ipfs-car
> Unused devDependencies
> * @types/pg
> Error: Process completed with exit code 255.
> ```
> 
> We should make sure these checks are passing before we merge this.

Glad to know its working as expected !,
I removed unused dependencies, Now it should not throw error.

juliangruber commented:
👏 👏 👏
